include/mercury/kw/APIConfig.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * \file APIConfig.h
+ *
+ * \copyright Copyright (C) 2019 Kano Computing Ltd.
+ *            License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+ *
+ * \brief Configuration file for web services
+ *
+ */
+
+#ifndef INCLUDE_MERCURY_KW_APICONFIG_H_
+#define INCLUDE_MERCURY_KW_APICONFIG_H_
+
+
+#include <yaml-cpp/yaml.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+
+namespace Mercury {
+namespace KanoWorld {
+
+
+/**
+ * \class APIConfig
+ * \brief Provides configurations for the API configurations
+ */
+class APIConfig {
+ public:
+    /**
+     * \brief Constructor
+     *
+     * \param config_files    (Optional) Config files to override defaults
+     */
+    explicit APIConfig(std::vector<std::string> config_files = {});
+
+    /**
+     * \brief Get the API URL
+     *
+     * \returns the configured API URL or https://worldapi.kano.me if one isn't
+     *          found
+     */
+    std::string get_api_url() const;
+
+    /**
+     * \brief Get a value from the configuration
+     *
+     * \param key    They value's key
+     *
+     * \returns the key's value
+     */
+    template<typename T>
+    T get_value(const std::string& key) const {
```

Comment:
Does the implementation of this function have to be in the header?

Reply:
Also, I could see a `has(const std::string& key)` method being useful here.

Reply:
Added